### ISSUE TITLE
Shared data not initialised by capture daemon

### ISSUE THREAD

**Issue by acrixl** [ZoneMinder/zoneminder#629]
Shared data not initialised by capture daemon, some query functions may not be available or produce invalid results for monitor internal

i do not want invalid results

so:

root@dvrsw:~# sysctl -a | grep -E "shmall|shmmax"
kernel.shmall = 18446744073692774399
kernel.shmmax = 536870912

root@dvrsw:~# ipcs

------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status  
0x01010f4d 0          root       600        1200712    4                       

------ Semaphore Arrays --------
key        semid      owner      perms      nsems  
0x00000000 0          www-data   600        1  
0x00000000 32769      www-data   600        1  
0x00000000 65538      www-data   600        1         

------ Message Queues --------
key        msqid      owner      perms      used-bytes   messages    

root@dvrsw:~# ipcs -m

------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status  
0x01010f4d 0          root       600        1200712    4                       

root@dvrsw:~# free
             total       used       free     shared    buffers     cached
Mem:       2047068     326492    1720576          0      15864     138472
-/+ buffers/cache:     172156    1874912
Swap:      3906556          0    3906556

**Comment by wwat** [comment]
@connortechnology Perhaps you should just fix the software then? Instead of wasting the time of people with such warnings or errors, you could spend that same CPU time to figure out which of the preconditions that should be true aren't and preferably make sure that at software release time the amount of such issues is zero.

**Comment by connortechnology** [comment]
@wwat I don't appreciate the tone of your message. 

Why don't you supply us with a PR doing what you suggest? Or put a bounty on it.

**Comment by wwat** [comment]
@connortechnology I am not the one publishing this software. If you (meaning whoever publishes this software in the end) cannot support its advertised features, you can also just withdraw the proposition. As such, the publishers of this software can both receive benefits (social benefits like "I am the software of some cool software package"), but they should also be able to receive criticism (i.e., "Your software sucks in ways X, Y and Z."). 

So, in short, I find your response childish. 

And, if you are asking whether I could do better, yes, I could. It's just that I have another job (where I *do* make sure that whatever I publish for clients works) and I don't particularly consider video monitoring an interesting problem domain.

**Comment by connortechnology** [comment]
@wwat since you don't like our releases, kindly stop using them and wasting our time. There shall be no further responses to you from me.

**Comment by attndeficit** [comment]
@connortechnology 

this RIGHT HERE is why open source software suffers.  His attitude is NOT the problem, you can easily ignore his "tone" (tone in text... lol are you a woman?)... you forget that these comments are google-facing and those of us doing our BEST TO SEARCH FOR HELP ON OUR OWN are constantly troubled by BS like this.  You helped nothing with your posts, connor and you could have easily NOT REPLIED.

I am troubleshooting your software on my own as we speak.  It frequently replaces ALL of my monitor feeds with grey feeds only.  Intermittently.  Sometimes it will go 72 hours without trouble, sometimes it will go 7 minutes then all feeds grey.  It's impossibly difficult to troubleshoot, it's even harder to search for answers with the attitudes of developers like YOU.  The only message of interest in my logs is the one referenced here so google BRINGS ME HERE and while it may make perfect sense to you (the one who has the intricate understanding of the package as a whole), it makes NO SENSE TO THE REST OF US who don't have several hours to try to figure out how the fuck you guys programmed all of this.  I could attempt to use my programming skills to go through the source and begin to solve it, but I don't have 100 hours for this, I need it to work now so I'm left with the only option of trying to find someone else who encountered the error and found a resolution.

I love this software for what it provides but in it's current form, it's a piece of shit that breaks constantly and needs a lot of work.  I'd love to bountysource but there's already over $2k on the x264 project alone and it doesn't seem to help progress development so why contribute to that?  I've got money, show me you're going to do it and i'll gladly PAY.

If you want to see more open source projects succeed, you, connortechnology, must change your attitude, the strangers in the wild who click sign-in and comment will NEVER BE CONTROLLABLE.

**Comment by knnniggett** [comment]
Alright, I've heard about enough.  This thread is being locked. 
Any additional whine threads are going to be deleted.
